examples/player/simid_player.js

```diff
@@ -240,67 +255,111 @@ class SimidPlayer {
+   * Checks whether the input dimensions are valid and fit in the player window.
+   * @param {!Object} dimensions A dimension that contains x, y, width & height fields.
+   * @return {boolean}
+   */
+  isValidDimensions_(dimensions) {
+    const playerDiv = document.getElementById('player_div');
+    const playerRect = playerDiv.getBoundingClientRect();
+
+    const heightFits = parseInt(dimensions.y) + parseInt(dimensions.height) <= parseInt(playerRect.height);
+    const widthFits = parseInt(dimensions.x) + parseInt(dimensions.width) <= parseInt(playerRect.width);
+    
+    return heightFits && widthFits;
+  }
+
+  /**
+   * Returns the specified dimensions of the non-linear creative.
    * @return {!Object}
    */
-  getNonLinearDimensions() {
-    const x_val = document.getElementById('x_val').value;
-    const y_val = document.getElementById('y_val').value;
-    const width = document.getElementById('width').value;
-    const height = document.getElementById('height').value;
+  getCreativeDimensions_() {
+    let newDimensions = {};
+    newDimensions.x = document.getElementById('x_val').value;
+    newDimensions.y = document.getElementById('y_val').value;
+    newDimensions.width = document.getElementById('width').value;
+    newDimensions.height = document.getElementById('height').value;
 
-    return {
-      'x' : x_val,
-      'y' : y_val,
-      'width' : width,
-      'height' : height,
-    };
+    return newDimensions;
+  }
+
+  /**
+   * Changes the simid iframe dimensions to the given dimensions
+   * @param {!Object} A dimension that contains an x,y,width & height fields.
+   */
+  setSimidIframeDimensions_(resizeDimensions) {
+    this.simidIframe_.style.height = resizeDimensions.height;
+    this.simidIframe_.style.width = resizeDimensions.width;
+    this.simidIframe_.style.left = `${resizeDimensions.x}px`;
+    this.simidIframe_.style.top = `${resizeDimensions.y}px`;
+  }
+
+  /** The creative wants to expand the ad. */
+  onExpandResize(incomingMessage) {
+    if (this.isLinearAd_) {
+      this.simidProtocol.reject(incomingMessage, "Cannot expand linear ads");
+      console.log("Cannot expand linear ads");
+  
+    } else {
+      const fullDimensions = this.getFullDimensions_(this.contentVideoElement_);
+      this.setSimidIframeDimensions_(fullDimensions);
+      
+      this.contentVideoElement_.pause();
+      this.simidProtocol.resolve(incomingMessage);
+    }
   }
 
   /**
    * Displays the non-linear creative with the specified size
    * on top of the video content.
    */
-  displayNonLinearCreative_() {
-    const dimensions = this.getNonLinearDimensions();
-    
-    this.simidIframe_.style.height = dimensions.height;
-    this.simidIframe_.style.width = dimensions.width;
-    this.simidIframe_.style.left = `${dimensions.x}px`;
-    this.simidIframe_.style.top = `${dimensions.y}px`;
+  displayNonLinearCreative(incomingMessage = CreativeMessage.REQUEST_COLLAPSE) {
```

Comment:
Don't use this constant for a default incoming message.

An incoming message is an object with a code that lets it know which message it's responding to.

I think this will all be solved once you implement an onRequestCollapse function.